src/main.rs

```diff
@@ -54,7 +57,49 @@ mod worker;
 
 pub type Result<T> = result::Result<T, Box<Error + Send + Sync>>;
 
+const CHILD_ENV_VAR: &'static str = "ripgrep-child-started";
+
+fn is_parent() -> bool {
+    env::var_os(CHILD_ENV_VAR).is_none()
+}
+
 fn main() {
+    if is_parent() {
+        use std::io::Write;
+        //Provides into_clonable method for Child struct.
+        //It allows to both wait and kill in parallel.
+        use clonablechild::{ChildExt};
+        use termcolor::{WriteColor, StandardStream};
+
+        //Stdout writer should be initialized prior to starting child process for reset to work.
+        let mut stdout = StandardStream::stdout(termcolor::ColorChoice::Auto);
+
+        let exe = env::current_exe().expect("Could not retrieve own exe's path");
+        let child = process::Command::new(exe).env(CHILD_ENV_VAR, "1")
+                                              .args(env::args().skip(1))
```

Comment:
You will want to use `env::args_os` here instead of `env::args` so that invalid unicode is still passed properly as an argument. Else, this will fail the `regression_210` test.